src/game/puppyprint.c

```diff
@@ -1267,10 +1268,10 @@ void puppyprint_print_deferred(void) {
             print_small_text(160, 80, "gEffectsMemoryPool is full.", PRINT_TEXT_ALIGN_CENTRE, PRINT_ALL, FONT_OUTLINE);
             return;
         }
-        s32 x = (sPuppyprintTextBuffer[i] << 8) & 0xFF;
-        x += sPuppyprintTextBuffer[i + 1] & 0xFF;
-        s32 y = (sPuppyprintTextBuffer[i + 2] << 8) & 0xFF;
-        y += sPuppyprintTextBuffer[i + 3] & 0xFF;
+        s32 x = ((sPuppyprintTextBuffer[i] << 8) & 0xFF) - 0x4000;
+        x += (sPuppyprintTextBuffer[i + 1] & 0xFF) - 0x40;
+        s32 y = ((sPuppyprintTextBuffer[i + 2] << 8) & 0xFF) - 0x4000;
+        y += (sPuppyprintTextBuffer[i + 3] & 0xFF) - 0x40;
```

Comment:
Not in a place to test, but this looks incorrect for multiple reasons.

1. `s32 x = ((sPuppyprintTextBuffer[i] << 8) & 0xFF) - 0x4000;` should be using `& 0xFF00`
2. `& 0xFF00` and `& 0xFF` should come after the subtraction I would think.

Even if these changes work, this still feels like a somewhat hacky solution. Still vouching for the struct method; shouldn't have any significant memory or performance impact either (might actually improve perf though)